crates/core_arch/src/mod.rs

```diff
@@ -66,6 +66,17 @@ pub mod arch {
         pub use crate::core_arch::riscv::*;
```

Comment:
If we're going to have separate intrinsics for riscv32 and riscv64 then we should probably rename the existing `riscv` module to `riscv32`. This makes it consistent with the other architectures.

Reply:
Yes, I'd accept this suggestion. RISC-V RV64 only contains more instructions in each set than RISC-V RV32, no instructions are removed from RISC-V RV32 yet. RV32 is not a legacy extension, it's better to make it explict. This way provides more readable rust docs as well.
After this change, I'd modify the implementation of `core::hint::spin_loop` in standard library.